### ISSUE TITLE
mirroring by tag does not

### ISSUE THREAD

**Issue by vrothberg** [containers/image#1043]
Reported by @praveenkumar.

A minimal reproducer (registries.conf):
```
[[registry]]
location = "docker.io/library/alpine:latest" 

[[registry.mirror]]
location = "docker.io/library/busybox"
```

... does not match when copying `docker.io/library/alpine:latest` while removing the ":latest" tag from the location (or prefix) will match and use busybox. The documentation clearly indicates that the tag should work.

**Comment by mtrmac** [comment]
Yes… OTOH, does that make sense? IMHO it fairly clearly does not make sense for digests.

As for tags, “freezing” a tag by replacing it with a digest would cause the mirror to be ignored, which would be surprising.

(Matching on a tag does not add significant functionality; we always try the primary location as well, so it is not _necessary_ to have tag matching to support partial mirrors that only contain some tags.)

---

Looking at the history, this also just seems to be bug in the documentation: the original mirroring documentation was not that explicit (#564), I have added the tag/digest mentions later in #617, and AFAICS it never worked; so the documentation is not _originally_ authoritative.

Actually, before mirrors existed at all, when `FindRegistry` was used only for TLS, it made lookups only using the repository name and not the tag/digest, starting in #468 back in 2018.

Cc: @saschagrunert .

**Comment by vrothberg** [comment]
I am cool to not allow tags/digests in the location/prefix but if we decide for that, it would be nice to throw a warning (or error out) if a location/prefix includes a tag/digest.

**Comment by mtrmac** [comment]
I have recently happened to implement that validation (along with more precise prefix matching) in the `remapIdentity` world; so we should probably shared that code between `policy.json` and `registries.conf`.

**Comment by zfrhv** [comment]
hello,
as i understand the feature to mirror by tag/digest doesnt works as mentioned in the docs
so why its still there?
should i make a pull request to remove this line: `host[:port]/namespace[/namespace…]/repo(:_tag|@digest)`?

**Comment by Noksa** [comment]
@zfrhv go ahead please. It is still the issue!

**Comment by zfrhv** [comment]
@Noksa 
Im actually not working with anything related to kubernetes/docker anymore (more than a year already).
I decided to study physics, idk if I will return to programming.
sorry
